pkg/controllers/dashboard/systemcharts/controller.go

```diff
@@ -110,3 +84,39 @@ func (h *handler) onRepo(key string, repo *catalog.ClusterRepo) (*catalog.Cluste
 
 	return repo, nil
 }
+
+func (h *handler) getChartsToInstall() []*chart.Definition {
```

Comment:
Can you elaborate on why you moved this to a function rather than leaving it as a var? Is it just a preference difference?